Add SONiC YANGs for Management Port, Management Interface, Management VRF and NTP.

bsun-sudo opened the pull request:
#### Why I did it
SONiC YANG model support for Management Port, Management Interface, Management VRF and NTP.

#### How I did it
Created sonic-xxx.yang files for these based on config-DB schema.
Added test cases for each.

#### How to verify it
Successfully build the following wheels,
make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl
make target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog
SONiC YANG model support for Management Port, Management Interface, Management VRF and NTP.

#### NOTE
Note that there is a build error for sonic_yang_mgmt-1.0-py3-none-any.whl if sample_config_db.json has a leaf with type boolean.
Tried true/"true"/True/"True", none works.

For now I have to remove this leaf in sample_config_db.json to the build to pass.

Need to have,

```
“MGMT_VRF_CONFIG”: {
    “vrf_global”: {
        “mgmtVrfEnabled”: “true”
    }
}
```

workaround,
```

“MGMT_VRF_CONFIG”: {
    “vrf_global”: { }
}

```

error log when use "true",

```
=================================== FAILURES ===================================
_____________________ Test_SonicYang.test_xlate_rev_xlate ______________________

self = <test_sonic_yang.Test_SonicYang object at 0x7f212d4fb748>
sonic_yang_data = {'syc': <sonic_yang.SonicYang object at 0x7f212e1f3780>, 'test_file': '../sonic-yang-models/tests/files/sample_config_db.json', 'yang_dir': '../sonic-yang-models/yang-models/'}

    def test_xlate_rev_xlate(self, sonic_yang_data):
        # In this test, xlation and revXlation is tested with latest Sonic
        # YANG model.
        test_file = sonic_yang_data['test_file']
        syc = sonic_yang_data['syc']
    
        jIn = self.readIjsonInput(test_file, 'SAMPLE_CONFIG_DB_JSON')
        jIn = json.loads(jIn)
        numTables = len(jIn)
    
        syc.loadData(jIn, debug=True)
        # check all tables are loaded and no tables is without Yang Models
        assert len(syc.jIn) == numTables
        assert len(syc.tablesWithOutYang) == 0
    
        syc.getData(debug=True)
    
        if syc.jIn and syc.jIn == syc.revXlateJson:
            print("Xlate and Rev Xlate Passed")
        else:
            print("Xlate and Rev Xlate failed")
            # print for better debugging, in case of failure.
            print("syc.jIn: {}".format({t:syc.jIn[t].keys() \
                for t in syc.jIn.keys()}))
            print("syc.revXlateJson: {}".format({t:syc.revXlateJson[t].keys() \
                for t in syc.revXlateJson.keys()}))
            # make it fail
>           assert False == True
E           assert False == True

tests/libyang-python-tests/test_sonic_yang.py:349: AssertionError
----------------------------- Captured stdout call -----------------------------
Read JSON Section: SAMPLE_CONFIG_DB_JSON
Xlate and Rev Xlate failed
```

error log when using "True",

```
def parse_data_mem(self, data, format, options=0):
>       return _yang.Context_parse_data_mem(self, data, format, options)
E       RuntimeError: Invalid value "True" in "mgmtVrfEnabled" element.

/usr/lib/python3/dist-packages/yang.py:2838: RuntimeError

```

joyas-joseph commented:
@praveen-li Could you please review this?

joyas-joseph commented:
```
diff --git a/src/sonic-yang-mgmt/sonic_yang_ext.py b/src/sonic-yang-mgmt/sonic_yang_ext.py
index b0122527..a4b502fe 100644
--- a/src/sonic-yang-mgmt/sonic_yang_ext.py
+++ b/src/sonic-yang-mgmt/sonic_yang_ext.py
@@ -477,6 +477,8 @@ class SonicYangExtMixin:
             vValue = list()
             for v in value:
                 vValue.append(_revYangConvert(v))
+        elif leafDict[key]['type']['@name'] == 'boolean':
+            vValue = 'true' if value else 'false'
         else:
             vValue = _revYangConvert(value)
 
```
This diff will fix the issue with using boolean values in sample_config_db.json

praveen-li commented:
> ```
> diff --git a/src/sonic-yang-mgmt/sonic_yang_ext.py b/src/sonic-yang-mgmt/sonic_yang_ext.py
> index b0122527..a4b502fe 100644
> --- a/src/sonic-yang-mgmt/sonic_yang_ext.py
> +++ b/src/sonic-yang-mgmt/sonic_yang_ext.py
> @@ -477,6 +477,8 @@ class SonicYangExtMixin:
>              vValue = list()
>              for v in value:
>                  vValue.append(_revYangConvert(v))
> +        elif leafDict[key]['type']['@name'] == 'boolean':
> +            vValue = 'true' if value else 'false'
>          else:
>              vValue = _revYangConvert(value)
>  
> ```
> 
> This diff will fix the issue with using boolean values in sample_config_db.json

A similar Fix is needed in  _findYangTypedValue(). Thanks.

config_db.json have input in String, So What are the values we are allowing to match with boolean true? or we can just have enumeration 'true' and 'false'

joyas-joseph commented:
> A similar Fix is needed in _findYangTypedValue(). Thanks.

When does this API get called?

> config_db.json have input in String, So What are the values we are allowing to match with boolean true? or we can just have enumeration 'true' and 'false'

"true"/"false".
We are seeing libyang error when tried with "True"
```
libyang[0]: Invalid value "True" in "mgmtVrfEnabled" element. (path: /sonic-mgmt-vrf:sonic-mgmt-vrf/MGMT_VRF_CONFIG/MGMT_VRF_CONFIG_LIST[name='vrf_global']/mgmtVrfEnabled)
```

praveen-li commented on src/sonic-yang-models/setup.py:
```diff
@@ -50,6 +50,10 @@
                          './yang-models/sonic-interface.yang',
                          './yang-models/sonic-loopback-interface.yang',
+                         './yang-models/sonic-mgmt-interface.yang',
+                         './yang-models/sonic-mgmt-port.yang',
```
sonic-mgmt-port ---> sonic-mgmt_port

bsun-sudo replied:
Done.

joyas-joseph replied:
We already have "sonic-loopback-interface.yang". Why the need for this change?

praveen-li reviewed (commented):
> > A similar Fix is needed in _findYangTypedValue(). Thanks.
> 
> When does this API get called?
> 
> > config_db.json have input in String, So What are the values we are allowing to match with boolean true? or we can just have enumeration 'true' and 'false'
> 
> "true"/"false".
> We are seeing libyang error when tried with "True"
> 
> ```
> libyang[0]: Invalid value "True" in "mgmtVrfEnabled" element. (path: /sonic-mgmt-vrf:sonic-mgmt-vrf/MGMT_VRF_CONFIG/MGMT_VRF_CONFIG_LIST[name='vrf_global']/mgmtVrfEnabled)
> ```

Not from Libyang point of view,  My question is more from user input POV,
Such as what user need to specify in config_db such as: "T"\"True"\"true"\"TRUE", i.e.  What values are processed by backend?

praveen-li commented on src/sonic-yang-models/tests/yang_model_tests/tests_config/mgmt_port.json:
```diff
@@ -0,0 +1,65 @@
+                        "admin_status": "up",
+                        "speed": 100,
+                        "autoneg": true,
+                        "description": "management interface",
```
This field is 'on'/'off' as per backend.

                /* Set autoneg and ignore the port speed setting */
                else if (fvField(i) == "autoneg")
                {
                    an = (fvValue(i) == "on");
                }

bsun-sudo replied:
@praveen-li . We use boolean for autoneg because the PORT_TABLE and MGMT_PORT_TABLE config DB schema use BIT for autoneg (https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md). 
I couldn't find your code, is it still in a PR? 

Thanks,

bsun-sudo replied:
Never mind. Got the recent commit for it from Joyas.

bsun-sudo replied:
Done. Changed to "on|off"

joyas-joseph commented:
> Not from Libyang point of view, My question is more from user input POV,
> Such as what user need to specify in config_db such as: "T""True""true""TRUE", i.e. What values are processed by backend?

"true"/"false" from user POV itself.

praveen-li commented:
> > Not from Libyang point of view, My question is more from user input POV,
> > Such as what user need to specify in config_db such as: "T""True""true""TRUE", i.e. What values are processed by backend?
> 
> "true"/"false" from user POV itself.

then we should use enumeration,  Allow only "true"/"false"....not "T"\"TRUE"\"True"

joyas-joseph commented:
> then we should use enumeration, Allow only "true"/"false"....not "T""TRUE""True"

boolean as per yang is "true" or "false" - https://tools.ietf.org/html/rfc6020#section-4.2.4

praveen-li commented:
> > then we should use enumeration, Allow only "true"/"false"....not "T""TRUE""True"
> 
> boolean as per yang is "true" or "false" - https://tools.ietf.org/html/rfc6020#section-4.2.4

 Not yang and Libyang POV, but how sonic backend processes it. Can you point to the code where the backend is processing the field for which you have used boolean. Thx.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-interface.yang:
```diff
@@ -0,0 +1,72 @@
+                        path /mgmtprt:sonic-mgmt-port/mgmtprt:MGMT_PORT/mgmtprt:MGMT_PORT_LIST/mgmtprt:name;
+                    }
+                }
+            }
```
This is for vrf association which is not applicable for mgmt intf. So maybe remove this?

bsun-sudo replied:
Removed.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-interface.yang:
```diff
@@ -0,0 +1,72 @@
+                    }
+                }
+
+                leaf ip_prefix {
```
add a  condition to make sure ip_prefix & gw are in the same addr family

bsun-sudo replied:
Modified Yang to match IPV4 addr with IPV4 GW addr and IPV6 addr with IPV6 GW addr. Added negative test cases as well.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-vrf.yang:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    container sonic-mgmt-vrf {
+        container MGMT_VRF_CONFIG {
```
try without list

bsun-sudo replied:
moved to container instead of list as there is only one mgmt vrf.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-ntp.yang:
```diff
@@ -0,0 +1,111 @@
+
+                leaf src {
+                    type union {
+                        type leafref {
```
add description for vrf which the intf belong to if possible

bsun-sudo replied:
Added description for NTP source interface for user configuration guideline.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-ntp.yang:
```diff
@@ -0,0 +1,111 @@
+                
+                key "global_key";
+
+                leaf global_key {
```
check if this can be a container.

bsun-sudo replied:
moved to container instead of list as there is only one global NTP configuration.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-ntp.yang:
```diff
@@ -0,0 +1,111 @@
+            description "NTP SERVER part of config_db.json";
+    
+            list NTP_SERVER_LIST {
+                max-elements 10;
```
add description

bsun-sudo replied:
Added description to include data point for numbers of NTP servers.

anshuv-mfst commented:
Hi @prsunny - could you please take a look, thanks.

bsun-sudo commented:
> > Not from Libyang point of view, My question is more from user input POV,
> > Such as what user need to specify in config_db such as: "T""True""true""TRUE", i.e. What values are processed by backend?
> 
> "true"/"false" from user POV itself.

@praveen-li 
For sonic-mgmt_vrf.yang, per your comment, I changed "list MGMT_VRF_CONFIG_LIST" to "container vrf_global", with only 1 leaf "mgmtVrfEnabled".  For this change, the json entry structure needs to contain at least one field in the sample_config_db.json. Otherwise I got a build error for sonic_yang_mgmt python wheel. This would require the fix for the boolean issue.

{
    “MGMT_VRF_CONFIG”: {
        “vrf_global”: {“mgmtVrfEnabled”: “true”}  <==== 
}

Thanks.

venkatmahalingam commented:
> > > then we should use enumeration, Allow only "true"/"false"....not "T""TRUE""True"
> > 
> > 
> > boolean as per yang is "true" or "false" - https://tools.ietf.org/html/rfc6020#section-4.2.4
> 
> Not yang and Libyang POV, but how sonic backend processes it. Can you point to the code where the backend is processing the field for which you have used boolean. Thx.

Praveen, as discussed in the call today, the backend uses 'true'/'false' for boolean, refer the backend code below https://github.com/Azure/sonic-buildimage/blob/master/files/image_config/interfaces/interfaces.j2

bsun-sudo commented on src/sonic-yang-models/tests/yang_model_tests/tests/mgmt_vrf.json:
```diff
@@ -0,0 +1,11 @@
+        "desc": "LOAD MGMT VRF TABLE WITH mgmtVrfEnabled SET TO FALSE SUCCESSFULLY. VERIFY mgmtVrfEnabled."
+    },
+    "MGMT_VRF_3_TEST": {
+        "desc": "LOAD MGMT VRF TABLE WITH DEFAULT mgmtVrfEnabled SUCCESSFULLY. VERIFY mgmtVrfEnabled."
```
Done.

bsun-sudo commented on src/sonic-yang-models/tests/yang_model_tests/tests/ntp.json:
```diff
@@ -0,0 +1,85 @@
+{
+    "NTP_SERVER_1_TEST": {
```
Done.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-port.yang:
```diff
@@ -0,0 +1,71 @@
+                    }
+                }
+
+                leaf autoneg {
```
Done.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-vrf.yang:
```diff
@@ -0,0 +1,43 @@
+            description "MANAGEMENT VRF part of config_db.json";            
+
+            list MGMT_VRF_CONFIG_LIST {
+                key "name";
```
Done.

bsun-sudo commented on src/sonic-yang-models/tests/yang_model_tests/tests_config/mgmt_interface.json:
```diff
@@ -0,0 +1,177 @@
+            }
+        }
+    },
+    "MGMT_INTERFACE_WITH_INVALID_IPV4": {
```
Done.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-port.yang:
```diff
@@ -0,0 +1,71 @@
+                }
+
+                leaf speed {
+                    type uint64 {
```
Done

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-port.yang:
```diff
@@ -0,0 +1,71 @@
+                }
+
+                leaf mtu{
+                    type uint32 {
```
Done.

bsun-sudo commented on src/sonic-yang-models/yang-models/sonic-mgmt-port.yang:
```diff
@@ -0,0 +1,71 @@
+
+                leaf speed {
+                    type uint64 {
+                        range "10|100|1000"; 
```
There was no specification at https://github.com/Azure/SONiC/wiki/Configuration. So we only allow user to configure the speeds that are applicable for mgmt port if autoneg is "off". As a random number within a range may not work. 
Addtional number may be added if applicable. Please let us know what do you suggest.

bsun-sudo commented:
@a-barboza Could you please review sonic-ntp.yang and the related test cases? Thanks.

a-barboza commented:
> @a-barboza Could you please review sonic-ntp.yang and the related test cases? Thanks.

@bsun-sudo Yes, gave the feedback

venkatmahalingam commented on src/sonic-yang-models/yang-models/sonic-mgmt_interface.yang:
```diff
@@ -0,0 +1,63 @@
+    description
+        "SONiC MANAGEMENT INTERFACE";
+
+    revision 2019-11-13 {
```
Please change the revision date.

bsun-sudo replied:
done.

venkatmahalingam commented on src/sonic-yang-models/yang-models/sonic-mgmt_interface.yang:
```diff
@@ -0,0 +1,63 @@
+                }
+
+                leaf-list forced_mgmt_routes {
+                    type stypes:sonic-ip-prefix;
```
Please add a description for this leaf-list.

bsun-sudo replied:
Done.

prsunny replied:
Is this covered? Just the IP?

```
            "forced_mgmt_routes": [
                "10.3.145.14",
```

venkatmahalingam commented on src/sonic-yang-models/yang-models/sonic-mgmt_port.yang:
```diff
@@ -0,0 +1,72 @@
+
+                leaf name {
+                    type string {
+                        pattern 'eth([1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[0-9])';
```
Cant we just have eth0?

bsun-sudo replied:
eth0 is included in the pattern.

venkatmahalingam commented on src/sonic-yang-models/yang-models/sonic-mgmt_port.yang:
```diff
@@ -0,0 +1,72 @@
+
+    description "MANAGEMENT PORT yang Module for SONiC OS";
+
+    revision 2019-11-13 {
```
Update the latest revision date.

bsun-sudo replied:
done.

venkatmahalingam commented on src/sonic-yang-models/yang-models/sonic-mgmt_port.yang:
```diff
@@ -0,0 +1,72 @@
+                leaf speed {
+                    type uint16 {
+                        range "10|100|1000"; 
+                    }
```
speed units? mbps?

bsun-sudo replied:
Added description, it is in megabytes

venkatmahalingam commented:
@bsun-sudo Please fix build errors.

bsun-sudo commented:
@venkatmahalingam @praveen-li Build error is because of the boolean in sample config file. https://dev.azure.com/mssonic/build/_build/results?buildId=11773&view=logs&j=5562265b-363f-5c64-5cff-7b508c285721&t=a6f2aaf4-4d30-50c0-c491-b0c5bd079658&l=788.

venkatmahalingam commented:
> @venkatmahalingam @praveen-li Build error is because of the boolean in sample config file. https://dev.azure.com/mssonic/build/_build/results?buildId=11773&view=logs&j=5562265b-363f-5c64-5cff-7b508c285721&t=a6f2aaf4-4d30-50c0-c491-b0c5bd079658&l=788.

OK, thanks Bing. @praveen-li Can we commit the fix for boolean in sample config file as part of this PR or you want to commit it in separate PR?

bhavini-gada commented:
@praveen-li , @prsunny - Could you please review all the updates from last weeks discussion ?

prsunny reviewed (approved):
lgtm

bhavini-gada commented:
@lguohan - This PR looks ready for merge. Can you please merge it ?

bsun-sudo commented:
@prsunny 

Is this covered? Just the IP?

            "forced_mgmt_routes": [
                "10.3.145.14",

I addressed your comments. Thanks.

bsun-sudo commented:
closed by mistake. Reopen PR>

bsun-sudo commented:
build error is not related to the changes in the PR.


        if (system(pyang_tree_cmd)):
            print("Failed: {}".format(pyang_tree_cmd))
>           exit(1)
E           SystemExit: 1

tests/test_sonic_yang_models.py:30: SystemExit
----------------------------- Captured stdout call -----------------------------
Failed: pyang -f tree ./yang-models/*.yang > ./yang-models/sonic_yang_tree
----------------------------- Captured stderr call -----------------------------
./yang-models/sonic-acl.yang:8: error: module "ietf-inet-types" not found in search path
./yang-models/sonic-acl.yang:17: warning: imported module "sonic-extension" not used
./yang-models/sonic-breakout_cfg.yang:8: warning: imported module "sonic-extension" not used
./yang-models/sonic-device_metadata.yang:8: error: module "ietf-yang-types" not found in search path
./yang-models/sonic-device_neighbor.yang:12: warning: imported module "sonic-extension" not used
./yang-models/sonic-interface.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-loopback-interface.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-port.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-portchannel.yang:13: warning: imported module "sonic-extension" not used
./yang-models/sonic-vlan.yang:17: warning: imported module "sonic-extension" not used
./yang-models/sonic-vrf.yang:5: warning: imported module "sonic-extension" not used

rathnasabapathyv commented:
@bsun-sudo : I see a recent conversation in sonic-community email thread about this recent compilation errors with sonic-acl.yang & sonic-device_metadata.yang. Looks like it's fixed in latest master. Can you try to sync the latest master to your branch and see whether builds succeeds?

**That community email conversation about this:**
![image](https://user-images.githubusercontent.com/55006784/116470125-d8d59900-a827-11eb-84ec-e2e1daa295c6.png)

venkatmahalingam commented on src/sonic-yang-models/tests/yang_model_tests/tests_config/ntp.json:
```diff
@@ -0,0 +1,265 @@
+			"Ethernet0",
+			"Vlan100",
+			"Loopback100",
+                        "PortChannel10",
```
Fix the alignment.

bsun-sudo replied:
done.

bhavini-gada commented:
@prsunny, @venkatmahalingam, @praveen-li - Could you please approve again ? So that this can be merged.

lguohan commented:
@praveen-li , can you check?

praveen-li commented on src/sonic-yang-mgmt/sonic_yang_ext.py:
```diff
@@ -477,6 +477,8 @@ def _revYangConvert(val):
             vValue = list()
             for v in value:
                 vValue.append(_revYangConvert(v))
+        elif leafDict[key]['type']['@name'] == 'boolean':
```
We need similar handling while converting from config DB to sonic Yang as well right, in below function:
_findYangTypedValue(self, key, value, leafDict)

bsun-sudo replied:
Hi Praveen,
For _findYangTypedValue(...),
def _findYangTypedValue(self, key, value, leafDict):

#```
        # convert config DB string to yang Type
        def _yangConvert(val):
            # Convert everything to string
            val = str(val)
            # find type of this key from yang leaf
            type = leafDict[key]['type']['@name']

            if 'uint' in type:
                vValue = int(val, 10)
            # TODO: find type of leafref from schema node
            elif 'leafref' in type:
                vValue = val
            #TODO: find type in sonic-head, as of now, all are enumeration
            elif 'head:' in type:
                vValue = val
            else:
                vValue = val
            return vValue
```
```
   when type is 'boolean', val itself is already 'true'. Please see below,
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> 
> /sonic/src/sonic-yang-mgmt/sonic_yang_ext.py(252)_yangConvert()
-> vValue = val
(Pdb) p key
'mgmtVrfEnabled'
(Pdb) p value
*** NameError: name 'value' is not defined
(Pdb) p leafDict
{'mgmtVrfEnabled': OrderedDict([('@name', 'mgmtVrfEnabled'), ('type', OrderedDict([('@name', 'boolean')])), ('default', OrderedDict([('@value', 'false')])), ('__isleafList', False)])}
(Pdb) p val
'true'
(Pdb) 

```

So here the change is not needed.

praveen-li commented on src/sonic-yang-models/yang-models/sonic-mgmt_port.yang:
```diff
@@ -0,0 +1,75 @@
+
+                leaf speed {
+                    type uint16 {
+                        range "10|100|1000"; 
```
By the way, from where we got this constraint on speed that it can take only these three values.

bhavini-gada replied:
These seem to be the supported speeds for mgmt interface currently

praveen-li commented on src/sonic-yang-models/yang-models/sonic-mgmt_vrf.yang:
```diff
@@ -0,0 +1,33 @@
+
+                leaf mgmtVrfEnabled {
+                    type boolean;
+                    default false;
```
We had a test case to verify that this default value is generated?

bsun-sudo replied:
That test case was removed based on the initial comments.
I've added the test case back now.

praveen-li commented on src/sonic-yang-models/yang-models/sonic-ntp.yang:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    import sonic-portchannel {
+        prefix po;
```
lag

bsun-sudo replied:
changed to lag.

praveen-li commented on src/sonic-yang-models/yang-models/sonic-ntp.yang:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    import sonic-port {
+        prefix prt;
```
port

bsun-sudo replied:
changed to port

bsun-sudo commented:
Build failure is due to "Port breakout failed" test case, not related to the change in this PR.

2021-05-13T16:55:47.5160738Z Invalid value "Ethernet8" in "port" element.
2021-05-13T16:55:47.5161028Z Failed to validate data tree
2021-05-13T16:55:47.5161318Z Data Validation Failed
2021-05-13T16:55:47.5161596Z [ERROR] Port breakout Failed!!! Opting Out  <<<<<<<<<<<<<<<<<
2021-05-13T16:55:47.5161906Z Failed to break out Port. Error:

bsun-sudo commented:
@praveen-li all comments are addressed or responded. Please review the change.

bsun-sudo commented:
/Azp run

rathnasabapathyv commented:
@praveen-li & @lguohan : 
Do you know any open PR to fix these DPB build fixes? This PR build is failing due to DPB, which is not relevant to this change. This PR is blocked due to just build failure

==============================================================================

tests/config_dpb_test.py:680: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def config_dpb_port8_4x25G_2x50G_f_l():
    
        # Input Data
        interface = 'Ethernet8'
        curMode = '4x25G[10G]'
        newMode = '2x50G'
    
        print("Mocked Child ports data-> {}".format([get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)]))
    
        config.get_child_ports = mock.MagicMock(
            side_effect = [get_child_ports_mock(interface, curMode), get_child_ports_mock(interface, newMode)])
    
        result = runner.invoke(config.config.commands["interface"].\
            commands["breakout"], ['{}'.format(interface), '{}'.format(newMode), '-v', '-f',\
             '-l', '-y'], obj=obj)
    
        print(result.exit_code, result.output)
        assert result.exit_code == 0
>       assert 'Breakout process got successfully completed.' in result.output
E       assert 'Breakout process got successfully completed.' in '\nRunning Breakout Mode : 4x25G[10G] \nTarget Breakout Mode : 2x50G\n\nPorts to be deleted : \n {\n    "Ethernet8": "...ate data tree\nData Validation Failed\n[ERROR] Port breakout Failed!!! Opting Out\nFailed to break out Port. Error: \n'
E        +  where '\nRunning Breakout Mode : 4x25G[10G] \nTarget Breakout Mode : 2x50G\n\nPorts to be deleted : \n {\n    "Ethernet8": "...ate data tree\nData Validation Failed\n[ERROR] Port breakout Failed!!! Opting Out\nFailed to break out Port. Error: \n' = <Result okay>.output

==============================================================================

prsunny commented:
/azp run

anshuv-mfst commented:
@judyjoseph / @prsunny - please help with merge, thanks

anshuv-mfst commented:
@bsun-sudo - please comment out the build failure causing changes and resubmit by 6/11.

bsun-sudo commented:
@anshuv-mfst 
"please comment out the build failure causing changes and resubmit by 6/11."
Just want to confirm you mean comment out "vlan related leafref" and "import sonic-vlan" for now, correct?

bhavini-gada commented:
/azpw run

bsun-sudo commented:
/azpw run

mssonicbld commented:
/AzurePipelines run

prsunny commented:
/azp run

bhavini-gada commented:
/azp run

bhavini-gada commented:
/azpw run

bhavini-gada commented:
@prsunny - Could you please approve & merge ?
